circle.yml

```diff
@@ -11,6 +11,10 @@ general:
     - http-servers/build/libs
     - ssl-config/build/libs
 
+dependencies:
+  override:
+    - ./gradlew resolveDependencies
```

Comment:
Yun was going to work on open sourcing a plugin that does this, fwiw.

Sent from my iPhone

On Apr 11, 2016, at 3:58 PM, Nick Miyake <notifications@github.com<mailto:notifications@github.com>> wrote:


In circle.yml<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_http-2Dremoting_pull_100-23discussion-5Fr59296203&d=CwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=stml6D2OZAkQr4h-02UjcVX5JFsoiD11C2l1QV5xgPE&s=les1ZQo--lmg5jwfURDLbdYFCz3So2clWsQQ0NmQTEM&e=>:

> @@ -11,6 +11,10 @@ general:
>      - http-servers/build/libs
>      - ssl-config/build/libs
>
> +dependencies:
> +  override:
> +    - ./gradlew resolveDependencies


Not quite, for a few reasons (also, I couldn't find assembleTest as a task, so assume you're referring to assemble).

First off, it actually doesn't download all the dependencies that resolveDependencies does. Compare the output of the build step using resolveDependencies (https://circleci.com/gh/palantir/http-remoting/135<https://urldefense.proofpoint.com/v2/url?u=https-3A__circleci.com_gh_palantir_http-2Dremoting_135&d=CwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=stml6D2OZAkQr4h-02UjcVX5JFsoiD11C2l1QV5xgPE&s=tdRX53A3r5yjSeqEva3AwhM9NysuTnq6IoaEpsnduh8&e=>) versus assemble (https://circleci.com/gh/palantir/http-remoting/136<https://urldefense.proofpoint.com/v2/url?u=https-3A__circleci.com_gh_palantir_http-2Dremoting_136&d=CwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=stml6D2OZAkQr4h-02UjcVX5JFsoiD11C2l1QV5xgPE&s=dhocCInJqz8ZhR_i3zN4XgtDin802atIFFrkuSLEiT0&e=>) -- the former properly grabs all dependencies so that the build step doesn't need to download anything further, while the latter does not.

RAINBOW has documented this issue further and suggests this work-around -- see issue #21<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_http-2Dremoting_pull_21&d=CwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=stml6D2OZAkQr4h-02UjcVX5JFsoiD11C2l1QV5xgPE&s=iAbSfvBtVBwbhiJIzGofp06GFGBH34-pvRcHSvuJFKs&e=> in RAINBOW's "build-system-issues" repository in GHE for more info.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_palantir_http-2Dremoting_pull_100_files_5224a2d46b443c85fcb5e852c1d2ed7be5411d34-23r59296203&d=CwMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=Vp81aOxWZvuVgVK_wp-VF3pIYG92B19LcCw6XKeYC0U&m=stml6D2OZAkQr4h-02UjcVX5JFsoiD11C2l1QV5xgPE&s=Vpn2o5QxnST_I_dRR2_CzIcHmYrUWEghSUK08F3rbqc&e=>